src/libopensc/card-myeid.c

```diff
@@ -131,9 +135,13 @@ static int myeid_init(struct sc_card *card)
 	u8 defatr[SC_MAX_ATR_SIZE];
 	size_t len = sizeof(defatr);
 	const char *atrp = myeid_atrs[MYEID_INFINEON_CHIP_ATR];
-
+	
 	LOG_FUNC_CALLED(card->ctx);
+
+	card->name = myeid_card_name;
```

Comment:
To be more clear, myeid_get_info adds version to card->name. There was a bug in a previous version, please see https://github.com/OpenSC/OpenSC/commit/fe2312dd1966f9fac6751345ab894b89b4f7df3c
Maybe you are right that we could do without global myeid_card_name (is card->name set already when myeid_init is called?), but card_name_buf cannot be local for this to work. This has to be tested if we modify it.